### PR TITLE
Remove using instanceof through use of polymorphism for events

### PR DESCRIPTION
**Pull request by vinopm**
Remove using instanceof through use of polymorphism for events and population passed in through constructor.

No intended functional changes.

Main changes you will see are that EventProcessor.java has now been renamed to EventProcessorRunner.java and specific Alert/Virus/Contact/Infection logic within EventProcessor has been moved over to the following 4 classes [AlertEventProcessor, InfectionEventProcessor, ContactEventProcessor, VirusEventProcessor]. The methods have just been copied/moved over with a few tweaks. Same with the tests - they have been moved over to 4 separate classes to match the source code structure

### CODE REVIEW

**Comment by sam-brett** [comment]
@vinopm You seem to be about 5 steps ahead of me at the moment. I was thinking about some of this on Saturday, checked my email and you'd done it. I'm going to start the review, but this week is looking nightmarish so bare with me.

File: src/main/java/uk/co/ramp/AppConfig.java [hunk 1]
```diff
@@ -16,9 +16,12 @@
 import uk.co.ramp.io.types.InputFiles;
 import uk.co.ramp.io.types.PopulationProperties;
 import uk.co.ramp.io.types.StandardProperties;
+import uk.co.ramp.people.Case;
+import uk.co.ramp.people.PopulationGenerator;
 import uk.co.ramp.utilities.UtilitiesBean;
 
 import java.io.*;
+import java.util.Map;
 import java.util.Optional;
```
Comment (sam-brett) [inline src/main/java/uk/co/ramp/AppConfig.java:25]: Remove unused imports

File: src/main/java/uk/co/ramp/event/processor/AlertEventProcessor.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import uk.co.ramp.utilities.MeanMax;
+
+import java.util.List;
+
+import static uk.co.ramp.people.AlertStatus.*;
+import static uk.co.ramp.people.AlertStatus.NONE;
+
+@Service
+public class AlertEventProcessor implements EventProcessor<AlertEvent> {
+    private static final Logger LOGGER = LogManager.getLogger(AlertEvent.class);
```
Comment (sam-brett) [inline src/main/java/uk/co/ramp/event/processor/AlertEventProcessor.java]:
```suggestion
    private static final Logger LOGGER = LogManager.getLogger(AlertEventProcessor.class);
```

File: src/main/java/uk/co/ramp/event/processor/CommonVirusEventProcessor.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import uk.co.ramp.people.Case;
+import uk.co.ramp.people.VirusStatus;
+import uk.co.ramp.utilities.MeanMax;
+
+import java.util.List;
+
+import static uk.co.ramp.people.VirusStatus.*;
+
+public abstract class CommonVirusEventProcessor<T extends Event> implements EventProcessor<T> {
+    private static final Logger LOGGER = LogManager.getLogger(InfectionEventProcessor.class);
```
Comment (sam-brett) [inline src/main/java/uk/co/ramp/event/processor/CommonVirusEventProcessor.java]:
```suggestion
    private static final Logger LOGGER = LogManager.getLogger(CommonVirusEventProcessor.class);
```

File: src/main/java/uk/co/ramp/event/processor/CommonVirusEventProcessor.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    private final DistributionSampler distributionSampler;
+
+    CommonVirusEventProcessor(Population population, DiseaseProperties diseaseProperties, DistributionSampler distributionSampler) {
+        this.population = population;
+        this.diseaseProperties = diseaseProperties;
+        this.distributionSampler = distributionSampler;
+    }
+
+    VirusStatus determineNextStatus(CommonVirusEvent event) {
+        Case thisCase = population.get(event.id());
```
Comment (sam-brett) [inline src/main/java/uk/co/ramp/event/processor/CommonVirusEventProcessor.java]: This is unused.

File: src/main/java/uk/co/ramp/utilities/UtilitiesBean.java [hunk 5]
```diff
@@ -10,19 +10,6 @@
 import java.util.stream.Stream;
```
Comment (sam-brett) [inline src/main/java/uk/co/ramp/utilities/UtilitiesBean.java]: Some unused imports.

Comment (sam-brett) [reply src/main/java/uk/co/ramp/utilities/UtilitiesBean.java]: Actually, it might be worth binning this class and having this as a static method on Case.

**Review by sam-brett — COMMENTED** [review]
That's all looking pretty good. I can't see any issues, but I'll give it a proper run through when I'm home.

File: src/main/java/uk/co/ramp/utilities/UtilitiesBean.java (hunk 5 above)
Comment (vinopm) [reply src/main/java/uk/co/ramp/utilities/UtilitiesBean.java]: Binned this class and moved the getMostSevere method into ContactEventProcessor as a private method, as it is only place it is used within the code base

**Pull request merged**